content/docs/admin/mail-config.md

```diff
@@ -0,0 +1,92 @@
++++
+title = "Mail Configuration"
+description = "Configuring Bookstack to send mail"
+date = "2017-08-08"
+type = "admin-docs"
+slug = "mail-config"
++++
+
+This document has been written by referencing the file at `config/mail.php`.
+
+Bookstack supports the following mechanisms for sending mail:
+
+1. SMTP;
+2. mail (php sendmail() function);
+3. sendmail (Linux sendmail);
+4. Mailgun (Rackspace Mailgun);
+5. Mandrill;
+6. Amazon SES ('ses'); or
+7. Log
+
```

Comment:
Just out of curiosity, what are these dependencies? Would be good to know what extra pieces are needed to get this to work reliably.

I've encountered environments where outbound SMTP traffic is actually rejected (even if it's going to an external gateway), which is why I wanted to document them.

Given that I've also got a pull request open for environment variables to be referenced in the config file (https://github.com/BookStackApp/BookStack/pull/468), would you be happy if I updated it with a list of the environment variables from that pull request instead?